File: uva/c_cpp/uva-12503.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    int ans,T,n,ins[1000],i,temp;
    char st[30];

    cin >> T;

    while(T--)
    {
        cin >> n;

        ans = 0;
        for(i = 1 ; i <= n ;i++)
        {
            scanf("%s",st);

            if(st[0] == 'L')
            {
                ans--;
                ins[i] = -1;
            }
            if(st[0] == 'R')
            {
                ans++;
                ins[i] = 1;
            }
            if(st[0] == 'S')
            {
                scanf("%s",st);
                scanf("%d",&temp);
                ans += ins[temp];
                ins[i] = ins[temp];
            }

        }

        cout << ans << endl;
    }
    return 0;
}

File: uva/java/uva_12503/src/uva_12503/Uva_12503.java
package uva_12503;

import java.util.Scanner;

public class Uva_12503 {

    /**
     * @param args the command line arguments
     */
    static Scanner scan = new Scanner(System.in);

    public static void main(String[] args) {
        // TODO code application logic here

        int ans, T, n, ins[] = new int[1000], i, temp;
        String st;
        //cin >> T;
        T = scan.nextInt();

        while (T-- > 0) {
            n = scan.nextInt();

            scan.nextLine();

            ans = 0;
            for (i = 1; i <= n; i++) {
                st = scan.nextLine();

                if (st.charAt(0) == 'L') {
                    ans--;
                    ins[i] = -1;
                }
                if (st.charAt(0) == 'R') {
                    ans++;
                    ins[i] = 1;
                }
                if (st.charAt(0) == 'S') {
                    st = scan.nextLine();
                    st = st.replace("SAME AS ","");
                    temp = new Integer(st);
                     System.out.println(2);
                    ans += ins[temp];
                    ins[i] = ins[temp];
                }
                    
                System.out.println(ans);
            }
            System.out.println(ans);
        }
    }

}
